homeassistant/components/airtouch4/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""Config flow for AirTouch4."""
+from airtouch4pyapi import AirTouch
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+def _createAirtouchObject(host):
+    return AirTouch(host)
+
+
+async def _validate_connection(hass: core.HomeAssistant, host):
+    airtouch = await hass.async_add_executor_job(_createAirtouchObject, host)
+
+    if hasattr(airtouch, "error"):
+        return airtouch.error
```

Comment:
If this is an exception, why not raise it?

Reply:
fair point - the library doesnt guarantee its an actual exception (was thinking it might be an error message sometimes/later) - have reworked this a bit to hopefully make more sense